.github/workflows/pypi-cd.yml

```diff
@@ -82,11 +71,29 @@ jobs:
               uses: actions/checkout@v4
               with:
                   submodules: "true"
+              
+              # For NPM and Cargo we need the the rc version to be {version}-rc{number} while for PyPi we need it to be {version}rc{number}
+              # This is because PyPi does not allow the '-' character in the version name
+              # So the tag will be as NPM and Cargo ask: v1.0.0-rc1 and for PyPi we'll chnage it to: v1.0.0rc1
+            - name: Change and set realease version for RC
+              if: ${{ github.event_name != 'pull_request' && contains(github.ref, 'rc') }}
+              shell: bash
+              run: |
+                  export version=`echo ${GITHUB_REF:11}`
+                  echo "RELEASE_VERSION=${version}" >> $GITHUB_ENV
+                  export py_version=`echo ${GITHUB_REF:11} | sed 's/-rc/rc/'`
+                  echo "PY_RELEASE_VERSION=${py_version}" >> $GITHUB_ENV
+
+            - name: Check that the release version dont have typo init
+              if: ${{ github.event_name != 'pull_request' && contains(github.ref, '-') && !contains(github.ref, 'rc') }}
+              run: exit 1
 
             - name: Set the release version
+              if: ${{ !contains(github.ref, 'rc') }}
```

Comment:
why don't we need this step for RC? 
and if we dont need it - why do you do `echo "PY_RELEASE_VERSION=${version}" >> $GITHUB_ENV` in line 96?